src/applications/timeouts.ts

```diff
@@ -162,7 +196,9 @@ export function reasonableTime(appOrParcel, lifecycle) {
   });
 }
 
-export function ensureValidAppTimeouts(timeouts) {
+export function ensureValidAppTimeouts(
+  timeouts: Partial<AppOrParcelTimeouts>
+): AppOrParcelTimeouts {
```

Comment:
Since Object.assign is a shallow merge of top-level properties, I don't think it would be equivalent to the existing functionality that is a deeper merge.

Here's the case where I think the behavior would be inequivalent:

```js
const timeouts = {
  mount: {
    millis: 2000
  }
};

const result = Object.assign({}, globalTimeoutConfig, timeouts);
// result is missing the dieOnTimeout and warningMillis properties
```

I will switch to using `Object.assign` rather than the IE-workaround `assign` function, but think the for loop should stay

Reply:
Good point. I typed this without really thinking about the structure of the timeouts objects.